src/parser/statement.rs

```diff
@@ -0,0 +1,103 @@
+use crate::lexer::Token;
+use crate::parser::ast::{Block, Expr, ExprKind, Statement};
+use crate::parser::error::ParseResult;
+use crate::parser::span::WithSpan;
+use crate::parser::Parser;
+use bumpalo::collections;
+
+impl<'s, 'a> Parser<'s, 'a> {
+    fn let_(&mut self) -> ParseResult<Statement<'s, 'a>> {
+        self.accept_required(tok![let])?;
+
+        let ident = self.ident()?;
+
+        let type_spec = if self.accept_optional(tok![:])?.is_some() {
+            Some(self.type_spec()?)
+        } else {
+            None
+        };
+
+        self.accept_required(tok![=])?;
+        let expr = self.expr()?;
+        self.accept_required(tok![;])?;
+
+        Ok(Statement::Let(ident, type_spec, self.alloc(expr)))
+    }
+
+    pub(super) fn block(&mut self) -> ParseResult<Block<'s, 'a>> {
+        let open_span = self.accept_required(Token::CurlyLeft)?;
+
+        let mut vec = collections::Vec::new_in(self.arena);
```

Comment:
Is the `collections::` intentional? I don't mind `use`ing `Vec` here directly.

Reply:
definitely not